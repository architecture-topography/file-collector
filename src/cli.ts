import chalk from "chalk";
import figlet from "figlet";
import commander from "commander";

const cli = (args: any = process.argv) => {
  console.log(
    chalk.yellow(
      figlet.textSync("TOPO JSON Collector", { horizontalLayout: "full" })
    )
  );

  commander
    .description("Please select one of the following actions: ")
    .option("-f, --file <file>", "Parse JSON file")
    .option("--host <host>", "Set host")
    .option("--port <port>", "Set port")
    .parse(process.argv);

  commander.outputHelp();
};

export default cli;
