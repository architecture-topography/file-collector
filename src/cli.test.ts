import cli from "./cli";
import { mockProcessStdout } from "jest-mock-process";
const CLIOUTPUT_INDEX = 2;

describe("cli", () => {
  let mockStdout: any;
  afterEach(() => {
    mockStdout.mockRestore();
  });

  beforeEach(() => {
    mockStdout = mockProcessStdout();
  });

  it("startup text should match snapshot", () => {
    cli();
    expect((mockStdout as any).mock.calls).toMatchSnapshot();
  });

  it("check that options are shown when no arguments are given", () => {
    cli();
    expect(JSON.stringify((mockStdout as any).mock.calls)).toContain('Usage:');
    expect(JSON.stringify((mockStdout as any).mock.calls)).toContain('Options:');
  })
});
