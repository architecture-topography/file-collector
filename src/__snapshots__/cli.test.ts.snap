// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cli startup text should match snapshot 1`] = `
Array [
  Array [
    "[999D[K",
  ],
  Array [
    "  [2mconsole.log[22m [2msrc/cli.ts:6[22m
    [33m  _____    ___    ____     ___          _   ____     ___    _   _      ____           _   _                 _                  [39m
    [33m |_   _|  / _ \\\\  |  _ \\\\   / _ \\\\        | | / ___|   / _ \\\\  | \\\\ | |    / ___|   ___   | | | |   ___    ___  | |_    ___    _ __ [39m
    [33m   | |   | | | | | |_) | | | | |    _  | | \\\\___ \\\\  | | | | |  \\\\| |   | |      / _ \\\\  | | | |  / _ \\\\  / __| | __|  / _ \\\\  | '__|[39m
    [33m   | |   | |_| | |  __/  | |_| |   | |_| |  ___) | | |_| | | |\\\\  |   | |___  | (_) | | | | | |  __/ | (__  | |_  | (_) | | |   [39m
    [33m   |_|    \\\\___/  |_|      \\\\___/     \\\\___/  |____/   \\\\___/  |_| \\\\_|    \\\\____|  \\\\___/  |_| |_|  \\\\___|  \\\\___|  \\\\__|  \\\\___/  |_|   [39m
    [33m                                                                                                                               [39m

",
    [Function],
  ],
  Array [
    "Usage: jest [options]

Please select one of the following actions: 

Options:
  -f, --file <file>  Parse JSON file
  --host <host>      Set host
  --port <port>      Set port
  -h, --help         output usage information
",
  ],
]
`;
